### PR TITLE
evm: Checked U256 casting

### PR DESCRIPTION
**Pull request by shohamc1**

### CODE REVIEW

File: lib/ain-evm/src/block.rs [hunk 1]
```diff
@@ -135,9 +135,9 @@ impl BlockService {
             .get_block_by_hash(&parent_hash)?
             .ok_or(format_err!("Parent block not found"))?;
         let parent_base_fee = parent_block.header.base_fee;
-        let parent_gas_used = parent_block.header.gas_used.as_u64();
+        let parent_gas_used = u64::try_from(parent_block.header.gas_used)?;
         let parent_gas_target =
-            parent_block.header.gas_limit.as_u64() / elasticity_multiplier.as_u64();
+            u64::try_from(parent_block.header.gas_limit)? / u64::try_from(elasticity_multiplier)?;
```
Comment (Jouzo) [inline lib/ain-evm/src/block.rs:140]: Don't we want to / before the casting here ?

File: lib/ain-evm/src/block.rs [hunk 2]
```diff
@@ -233,10 +236,10 @@ impl BlockService {
                 }
 
                 let mut block_rewards = Vec::new();
-                let priority_fees = block_eip_tx
+                let priority_fees: Vec<f64> = block_eip_tx
```
Comment (Jouzo) [inline lib/ain-evm/src/block.rs:239]: Can type infer here

**Comment by Jouzo** [comment]
Missing the checked_ops in `base_fee_calculation`

File: lib/ain-evm/src/block.rs [hunk 3]
```diff
@@ -189,11 +215,14 @@ impl BlockService {
...
+                    u64::try_from(block.header.gas_used)? as f64
+                        / u64::try_from(block.header.gas_limit)? as f64
                 };
 
-                (base_fee, gas_ratio)
+                Ok((base_fee, gas_ratio))
             })
+            .collect::<Result<Vec<(U256, f64)>>>()?
+            .into_iter()
             .unzip();
```
Comment (Jouzo) [inline lib/ain-evm/src/block.rs:226]: Could `fold` instead of collect -> into_iter -> unzip

Comment (shohamc1) [reply lib/ain-evm/src/block.rs]: Fixed: 88e528c040425f08134ca14bf731561f55a3234f

**Review by prasannavl — APPROVED** [review]

**Pull request merged**